spec/classes/agent_spec.rb

```diff
@@ -68,24 +68,21 @@
           end
         else
           it do
-            is_expected.to contain_package(package_name).with(
-              ensure:   'present',
-              require:  'Class[Zabbix::Repo]',
-              tag:      'zabbix'
-            )
+            is_expected.to contain_package(package_name).
+              with_ensure('present').
+              with_tag('zabbix').
+              that_requires('Class[zabbix::repo]')
           end
           it do
-            is_expected.to contain_service(service_name).with(
-              ensure:     'running',
-              enable:     true,
-              hasstatus:  true,
-              hasrestart: true,
-              require:    "Package[#{package_name}]"
-            )
+            is_expected.to contain_service(service_name).
+              with_ensure('running').
+              with_enable(true).
+              with_service_provider(facts[:osfamily] == 'AIX' ? 'init' : nil).
+              that_requires("Package[#{package_name}]")
           end
 
           it { is_expected.to contain_file(include_dir).with_ensure('directory') }
-          it { is_expected.to contain_zabbix__startup(service_name).with(require: "Package[#{package_name}]") }
```

Comment:
This `with_require` style happens more often in this module. I just fixed it for this file

Reply:
let's open an issue as reminder to do it in a next iteration

Reply:
what's bad about it?

Reply:
Consider this code:
```puppet
class myclass {
  package { 'mypackage':
  }

  file { '/etc/mypackage.conf':
    require => Package['mypackage'],
  }

  service { 'mypackage':
    subscribe => File['/etc/mypackage.conf'],
  }
}
```

Then the test code:
```ruby
it { is_expected.not_to contain_service('mypackage').with_require('Package[mypackage]') }
it { is_expected.to contain_service('mypackage').that_requires('Package[mypackage]') }
```

What I'm trying to show is that one tests for a parameter on a resource and the other the relationship exists in the catalog. That takes transitive requirements into account.